CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
+    $apiRequest = (array) $e->getApiRequest();
+    if (!self::shouldRun($apiRequest)) {
+      return;
+    }
+
+    $handler = CaseTypeCategoryEventHandlerFactory::create();
+    $handler->onUpdate(
+      $apiRequest['params']['id'],
+      isset($apiRequest['params']['is_active']) ? $apiRequest['params']['is_active'] : NULL,
+      isset($apiRequest['params']['icon']) ? $apiRequest['params']['icon'] : NULL
+    );
+  }
+
+  /**
+   * Determines if the processing will run.
+   *
+   * @param array $apiRequest
+   *   Api request data.
+   *
+   * @return bool
+   *   TRUE if processing should run, FALSE otherwise.
+   */
+  protected static function shouldRun(array $apiRequest) {
```

Comment:
Can we make all the protected functions in this class to be private ? Sorry didn't catch this before